src/pages/Account/Tabs/ModulesTab/Contract.tsx

```diff
@@ -450,11 +456,70 @@ function ReadContractForm({
 
   const onSubmit: SubmitHandler<ContractFormType> = async (data) => {
     logEvent("read_button_clicked", fn.name);
-    const viewRequest: Types.ViewRequest = {
-      function: `${module.address}::${module.name}::${fn.name}`,
-      type_arguments: data.typeArgs,
-      arguments: data.args,
-    };
+    let viewRequest: Types.ViewRequest;
+    try {
+      viewRequest = {
+        function: `${module.address}::${module.name}::${fn.name}`,
+        type_arguments: data.typeArgs,
+        arguments: data.args.map((arg, i) => {
+          if (fn.params[i].includes("vector")) {
+            // encode the vector
+            const rawVector = arg.trim().slice(1, -1).split(",");
```

Comment:
its gonna break if vector<u8> is inputed as "2,3" instead of "[2,3]". lets be a bit flexible here to cover different cases.

Reply:
fixed